projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -428,10 +440,19 @@ public static function get_blog_id() {
 	/**
 	 * Get the id of the connected payment acount (Stripe etc).
 	 *
-	 * @return int|void
+	 * @return bool
 	 */
-	public static function get_connected_account_id() {
-		return get_option( self::$connected_account_id_option_name );
+	public static function has_connected_account() {
+
+		// This is the primary solution.
+		$has_option = get_option( self::$has_connected_account_option_name, false ) ? true : false;
+		if ( $has_option ) {
+			return true;
+		}
+
+		// This is the fallback solution.
+		// TODO: Remove this once the has_connected_account_option is migrated to all sites.
+		return get_option( 'jetpack-memberships-connected-account-id', false ) ? true : false;
 	}
```

Comment:
@pottedmeat 

```suggestion
	public static function has_connected_account() {

		// This is the primary solution.
		$has_option = get_option( self::$has_connected_account_option_name, null );
		if ( ! is_null( $has_option ) ) {
			return true;
		}

		// This is the fallback solution.
		// TODO: Remove this once the has_connected_account_option is migrated to all sites.
		return get_option( 'jetpack-memberships-connected-account-id', false ) ? true : false;
	}
```

Is this what you were thinking?  I don't like the ambiguity with what the option _could be_.  I think I would rather be more explicit. Also it seems that if the option was set as false this would still return true, but that's probably my implementations fault.

Reply:
You'd still need to do: `return $has_option ? true : false;` within that if block.